google/cloud/aiplatform/base.py

```diff
@@ -607,6 +608,10 @@ def _assert_gca_resource_is_available(self) -> None:
     def __repr__(self) -> str:
         return f"{object.__repr__(self)} \nresource name: {self.resource_name}"
 
+    def to_dict(self) -> dict:
```

Comment:
Please add a test case invoking this method.

Reply:
I think the return type can be further qualified `Dict[str, Any]`

Reply:
Do you think it's a good idea to replace some of the MessageToDict usage cases with this new method? Then it will be auto-tested. https://github.com/googleapis/python-aiplatform/search?q=MessageToDict

Reply:
Done.
I've switched .metadata property to use `.to_dict()` and also switched one of the dataset tests.